synapseclient/core/credentials/credential_provider.py

```diff
@@ -77,6 +77,7 @@ def _create_synapse_credential(
             profile = syn.restGET("/userProfile", auth=credentials)
             profile_username = profile.get("userName")
             profile_emails = profile.get("emails", [])
+            profile_displayname = profile.get("displayName")
```

Comment:
From the Rest API docs:

> displayName 	STRING 	This field is deprecated and will always be null.

https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/UserProfile.html

Is this still being set?

Reply:
This would need to follow the same check that existed in the client.py script before:

```
            display_name = (
                profile["displayName"]
                if "displayName" in profile
                else self.credentials.username
            )
```

I think:
`profile_displayname = profile.get("displayName", profile_username)`

That is assuming that `displayName` has cases where it is not None. If it is not None then I think we can revert most of these changes and instead just use username in the welcome message:
`self.logger.info(f"Welcome, {self.credentials.username}!\n")`

Reply:
I did test this out and from that API I did get back a `displayName` for my account:

`"displayName": "Bryan Fauble",`

So I think the documentation for that API is incorrect. Could you create a PLFM jira ticket and mention this? The docs should be corrected, or the field should indeed return null.

Reply:
Thanks for catching this. I got the displayName from my profile as well. I will add the check back.